### PR TITLE
Adding server core module.

### PR DESCRIPTION
**Pull request by vasap87**

### CODE REVIEW

File: server-core/src/main/java/com/stnetix/ariaddna/servercore/tools/ServiceTree.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    private HashSet<ServiceNode> nodes;
+    private ArrayList<ServiceNode> startQueue;
+
+    private static AriaddnaLogger logger = AriaddnaLogger.getLogger(ServiceTree.class);
+
+    public ServiceTree() {
+        nodes = new HashSet<>();
+    }
+
+    public void addService(IService service, IService parent) throws ServiceNotFoundException {
```
Comment (kosmoflyko) [inline server-core/src/main/java/com/stnetix/ariaddna/servercore/tools/ServiceTree.java]: What we do if service has more one dependencies?

File: server-core/src/main/java/com/stnetix/ariaddna/servercore/tools/ServiceTree.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+                logger.info("Service "+node.getService().getClass().getTypeName() + " started by server-core service.");
+            }
+        }
+    }
+
+    public void monitoring() throws AriaddnaException {
+        boolean isStoped = false;
+        while (!isStoped) {
+            for (ServiceNode node : startQueue) {
+                if (!node.getService().isRun()) {
```
Comment (kosmoflyko) [inline server-core/src/main/java/com/stnetix/ariaddna/servercore/tools/ServiceTree.java]: Why monitoring don't logging services in "run" state

File: server-core/src/main/java/com/stnetix/ariaddna/servercore/tools/ServiceTree.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+            nodes.add(node);
+        }
+    }
+
+    public void startServer() throws AriaddnaException {
+        startQueue = createStartQueue();
+        for (ServiceNode node : startQueue) {
+            if (!node.getService().isRun()) {
+                Thread thread = new Thread(node.getService());
+                thread.start();
```
Comment (kosmoflyko) [inline server-core/src/main/java/com/stnetix/ariaddna/servercore/tools/ServiceTree.java]: It's bad practice run uncontrol tread. Please use PoolExecuter and save references to runnig thread.

File: server-core/src/main/java/com/stnetix/ariaddna/servercore/tools/ServiceTree.java (hunk 1 above)
Comment (vasap87) [inline server-core/src/main/java/com/stnetix/ariaddna/servercore/tools/ServiceTree.java]: We add this service with each parent. startServer() method check if !node.getService().isRun().

File: server-core/src/main/java/com/stnetix/ariaddna/servercore/tools/ServiceTree.java (hunk 3 above)
Comment (vasap87) [inline server-core/src/main/java/com/stnetix/ariaddna/servercore/tools/ServiceTree.java]: I add ExecutorService to fields of ServiceTree class.

File: server-core/src/main/java/com/stnetix/ariaddna/servercore/tools/ServiceTree.java (hunk 2 above)
Comment (vasap87) [inline server-core/src/main/java/com/stnetix/ariaddna/servercore/tools/ServiceTree.java]: Done

File: server-core/src/main/java/com/stnetix/ariaddna/servercore/tools/ServiceTree.java (hunk 1 above)
Comment (kosmoflyko) [inline server-core/src/main/java/com/stnetix/ariaddna/servercore/tools/ServiceTree.java]: You did not understand me. In current code you use serial initialization. It is true if own service have only one parent, for example: server-core depends on common-utils.
But we will have services with many dependencies, for example: VUFS Synchronization Service depends on Balancer Service and Allocation Storage Service, but Balancer Service and Allocation Storage Service is independent services.

Comment (slonikmak) [inline server-core/src/main/java/com/stnetix/ariaddna/servercore/tools/ServiceTree.java]: maybe it can be solved using spring?

**Comment by kosmoflyko** [comment]
fix #18

**Pull request merged**